gulp/tasks/lint.js

```diff
@@ -4,16 +4,21 @@ var gulp = require( 'gulp' );
 var plugins = require( 'gulp-load-plugins' )();
 var configLint = require( '../config' ).lint;
 var handleErrors = require( '../utils/handle-errors' );
+var minimist = require( 'minimist' );
 
 /**
  * Generic lint a script source.
  * @param {string} src The path to the source JavaScript.
  * @returns {Object} An output stream from gulp.
  */
 function _genericLint( src ) {
-  return gulp.src( src )
-    .pipe( plugins.eslint() )
+  // Grab the --fix flag from the command-line, if available.
+  var commandLineParams = minimist( process.argv.slice( 2 ) );
+  var willFix = commandLineParams.fix || false;
+  return gulp.src( src, { base: './' } )
+    .pipe( plugins.eslint( { fix: willFix } ) )
     .pipe( plugins.eslint.format() )
+    .pipe( gulp.dest( './' ) )
```

Comment:
@Scotchester `gulp.dest` in this instance isn't the same as the chained variable naming we've been using like `styles.dest`, it's a method for writing to disk.